### ISSUE TITLE
[feature request] support of wiktionary.org dictionaries

### ISSUE THREAD

**Issue by MR0205** [ninja33/ODH#94]
Hey guys, the ODH extension is super useful and easy to use. I was looking for a way to use it with wiktrionary.org, but I am not enough tech savy to handle all this JS code to add a good dictionary parser on myself. Is there any chance you will add support of wiktrionary.org in near future?

Wiktionary.org provides the defnitions of words in the same language, there are dictionaries for many many languages. In my opinion the quality of the dictionaries is superb and implementing a parser just for one language  would automatically apply to all other languages of wiktionary.org.

Cheers!

**Comment by ninja33** [comment]
Well, the popup window is just with 400x300 height and width.  It is not readable that puts too much information in that window.  So, instead of showing whole wiktionary page there, you must figure out what information you want to display(or export to Anki). 

For example, I tried to search English word 'nation' in this url: https://en.wiktionary.org/wiki/nation
There are so many fields, like Pronunciation, Etymology , Noun, Usage notes, Synonyms
You dont want to put all these information in a small popup windows, right? So, you have to figure out which fields you want in specified language. 

That's why JS code introducted to parse that page. but, not every page is same for different languages. for Example, for Chinese language, there is no IPA information and inflection. That's why each JS code parser is specified, and only that language learner knows what he/she wants.

You can leave a message here, tell me which language and which fields in that language you want? also show me some example words, I will check if it's workable.

**Comment by MR0205** [comment]
Yes, the pop-up window size is a bit limiting in terms of for what use it is mostly suited. I use the plugin as a one-click note export tool, and less as a dictionary itself, so the size being small doesn't matter that much. Probably if it woudln't make the addon more complex you could allow the popup window to extend vertically to display more content of a dictionary note (for a 80% of the web-browser window or so).

The wiktionary pages indeed display tons of information, that is why a parser would also help to navigate faster instead of being forced to scroll down for a small piece of information that is just happened to be at the bottom.

Naturally, one improvement to the parser and note exporter mechanism that comes to my mind is that the pop-up could intentionally omit certain fields and dicide not to display them (e.g. example sentences) to make a pop-up bubble smaller. Whereas the export function would still transfer to Anki all the information and let the user utilise it on their front and back cards.

I am interested mostly in French https://fr.wiktionary.org/ and the most important parts of speech are:
nouns, adjectives, verbs and adverbs. Here are the examples:
- noun [https://fr.wiktionary.org/wiki/nation](https://fr.wiktionary.org/wiki/nation)
- adjective [https://fr.wiktionary.org/wiki/vert](https://fr.wiktionary.org/wiki/vert)
- verb [https://fr.wiktionary.org/wiki/comprendre](https://fr.wiktionary.org/wiki/comprendre)
- adverb [https://fr.wiktionary.org/wiki/vite](https://fr.wiktionary.org/wiki/vite)

The parser would need to choose the language (that is French for a french parser) because on the same page the same word is expalined for many languages; and also parser would need to go trough all the parts of speech (see vert, or vite words which both have multiple parts of speech definitions). As of the granularity of the exported definition I would split the dictionary articles on the level of different meanings, and not on the level of parts of speech. (It is my personal preference in laguage learning).

For the fields:
- expression, i.e. the word or phrase
- audio (optional)

and for each parts of speech in the list:
- extra info: pronounciation, gender for nouns and transitivity for verbs.
- definitions to choose from
for each definition
- sentences (hidden in the bubble, but exported into Anki)
